runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -0,0 +1,65 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Message;
+
+import org.corfudb.runtime.CorfuOptions;
+import org.corfudb.runtime.collections.CorfuTable.Index;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Optional;
+
+/**
+ * Created by hisundar on 2019-08-12.
+ */
+public class ProtobufIndexer implements CorfuTable.IndexRegistry<Message, CorfuRecord<Message, Message>> {
+
+    private HashMap<String,
+            CorfuTable.Index<Message, CorfuRecord<Message, Message>, ? extends Comparable<?>>>
```

Comment:
looks really overcomplicated

Reply:
It is literally the CorfuTable's Indexer. No added complexity outside of it.